File: csharp/version1/Program.cs
public static class BlackJack {

    private static Random rng = new Random();  

    public static void Shuffle<T>(this IList<T> list)  
    {  
        int n = list.Count;  
        while (n > 1) {  
            n--;  
            int k = rng.Next(n + 1);  
            T value = list[k];  
            list[k] = list[n];  
            list[n] = value;  
        }  
    }
    private static List<string> suits = new List<string>(){"S","D","C","H"};
    private static List<string> values = new List<string>(){"A","1","2","3","4","5","6","7","8","9","J","Q","K"};

    public static List<string> NewDeck() {
        var my_deck = new List<string>();
        suits.ForEach((e) => values.ForEach((e2) => my_deck.Add(e2+e)));
        return my_deck;
    }

    public static void ShuffleDeck(in List<string> card_deck) {
        card_deck.Shuffle<String>();
    }

    public static void DealCards(in List<string> my_deck) {
        var choice_one = my_deck[0];
        my_deck.RemoveAt(0);
        var choice_two = my_deck[0];
        my_deck.RemoveAt(0);
        Console.WriteLine($"{choice_one}, {choice_two}, {my_deck.Count}");
    }
}


public class Program {
    public static void Main() {
        Console.WriteLine("Generating a new deck.");
        for(int x = 0; x < 100; x++) {
            var card_deck = BlackJack.NewDeck();
            for(int i = 0; i < 26; i++) {
                BlackJack.ShuffleDeck(card_deck);
                BlackJack.DealCards(card_deck);
            }
        }
    }
}

File: vb/version1/Program.vb
Imports System

Module Program
    Sub Main(args As String())
        Console.WriteLine("Generating a new deck.")
        Dim card_deck As List(Of String)
        Dim x As Integer = 0
        Dim i As Integer = 0
        For x = 0 to 99
            card_deck = NewDeck()
            For i = 0 to 25
                ShuffleDeck(card_deck)
                DealCards(card_deck)
            Next
        Next
    End Sub

    Function NewDeck() As List(Of String)
        Dim my_deck As New List(Of String)
        Dim suits As New List(Of String) From {"S","D","C","H"}
        Dim values As New List(of String) From {"A","1","2","3","4","5","6","7","8","9","J","Q","K"}
        For Each s As String In Suits
            For Each v As String In values
                my_deck.Add(v+s)
            Next
        Next
        Return my_deck
    End Function

    Sub ShuffleDeck(ByRef card_deck As List(Of String))
        Dim r As New Random()
        Dim n As Integer = card_deck.Count
        While n > 1
            n = n-1
            Dim k As Integer = r.Next(n + 1)
            Dim value As String = card_deck(k)
            card_deck(k) = card_deck(n)
            card_deck(n) = value
        End While
    End Sub

    Sub DealCards(ByRef my_deck As List(Of String))
        Dim choice_one As String = my_deck(0)
        my_deck.RemoveAt(0)
        Dim choice_two As String = my_deck(0)
        my_deck.RemoveAt(0)
        Console.WriteLine($"{choice_one}, {choice_two}, {my_deck.Count}")
    End Sub
End Module
